### PR TITLE
Move testing related classes from src to tests, add deprecations

### PR DESCRIPTION
**Pull request by keradus**
Deps:
- [ ] #2897

### CODE REVIEW

File: tests/Test/IntegrationCase.php [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests\Test;
+
+use PhpCsFixer\RuleSet;
+
+/**
+ * @author Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ */
```
Comment (SpacePossum) [inline tests/Test/IntegrationCase.php]: `@internal`

File: tests/Test/AbstractIntegrationTestCase.php [hunk 2]
```diff
@@ -0,0 +1,371 @@
...
+ * {"php": 50600**}
+ * --EXPECT--
+ * Expected code after fixing
+ * --INPUT--*
+ * Code to fix
+ *
+ *   * Section or any line in it may be omitted.
+ *  ** PHP minimum version. Default to current running php version (no effect).
+ *
+ * @author SpacePossum
```
Comment (SpacePossum) [inline tests/Test/AbstractIntegrationTestCase.php]: `@internal`

File: tests/Test/AbstractFixerTestCase.php [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+use PhpCsFixer\RuleSet;
+use PhpCsFixer\Tests\Test\Assert\AssertTokensTrait;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+use PhpCsFixer\Utils;
+use PHPUnit\Framework\TestCase;
+use Prophecy\Argument;
+
+/**
+ * @author Dariusz Rumiński <dariusz.ruminski@gmail.com>
```
Comment (SpacePossum) [inline tests/Test/AbstractFixerTestCase.php]: `@internal`

File: tests/Test/AbstractIntegrationTestCase.php (hunk 2 above)
Comment (keradus) [inline tests/Test/AbstractIntegrationTestCase.php]: vide #2896

**Comment by keradus** [comment]
Used during merging #2896 into 2.4 line

**Pull request closed without merging**